common/djangoapps/track/backends/mongodb.py

```diff
@@ -46,6 +47,8 @@ def __init__(self, **kwargs):
         db_name = kwargs.get('database', 'track')
         collection_name = kwargs.get('collection', 'events')
 
+        auth_source = kwargs.get('auth_source', '') or None
```

Comment:
```suggestion
        auth_source = kwargs.get('auth_source') or None
```

Reply:
Done.

Reply:
Yup, now I remember why I added that fallback parameter. When you call .pop without a fallback param, it throws an error if the key is absent. Adding it back where needed.